### ISSUE TITLE
get_object_stream() no longer accepts parameter writer

### ISSUE THREAD

**Issue by lyokha** [durch/rust-s3#329]
**Describe the bug**

This is not actually a bug, but rather lack of functionality in the current beta release that was available in the stable release *0.32.3*. Particularly, it seems that it is no longer possible to use `tokio::net::UnixStream` with `get_object_stream()`.

I need this removed functionality to make *rust-s3* inter-operate with an older C++ *boost::asio* code. C++ cannot use async Rust functions directly, therefore the code leverages `UnixStream`s that are treated asynchronously at the kernel level.

Below is an excerpt of the affected code.

```rust
#[no_mangle]
#[allow(clippy::missing_safety_doc)]
pub unsafe extern "C" fn c_init_object_stream(rt: *const Runtime, write: c_int,
    path: *const c_char) -> *mut StreamHandle
{
    if path.is_null() {
        return std::ptr::null_mut()
    }

    let rt_ref = rt.as_ref().unwrap();

    let (client, server) = init_object_stream();
    let fd = client.as_raw_fd();
    let bucket = get_bucket();
    let path = unsafe { CStr::from_ptr(path).to_str().unwrap().to_owned() };
    let join_handle = rt_ref.spawn(async move {
        let mut server = UnixStream::from_std(server).unwrap();
        let res = if write == 0 {
            get_object(bucket, &mut server, &path).await
        } else {
            put_object(bucket, &mut server, &path).await
        };
        res
    });
    Box::into_raw(Box::new(StreamHandle{
        rt_handle: rt, join_handle: Box::new(join_handle), client: Box::new(client), fd }))
}
```

where

```rust
type StdUnixStream = std::os::unix::net::UnixStream;

fn init_object_stream() -> (StdUnixStream, StdUnixStream) {
    let (client, server) = StdUnixStream::pair().unwrap();
    client.set_nonblocking(true).unwrap();
    server.set_nonblocking(true).unwrap();
    (client, server)
}

async fn put_object(bucket: &Bucket, server: &mut UnixStream, path: &str) -> Result<u16, S3Error>
{
    let status = bucket.put_object_stream(server, path).await;
    let status_code = match status {
        Ok(status) => status,
        Err(S3Error::Http(status, _body)) => status,
        Err(_) => 500
    };
    Ok(status_code)
}

async fn get_object(bucket: &Bucket, server: &mut UnixStream, path: &str) -> Result<u16, S3Error>
{
    let status = bucket.get_object_stream(path, server).await;
    let status_code = match status {
        Ok(status) => status,
        Err(S3Error::Http(status, _body)) => status,
        Err(_) => 500
    };
    Ok(status_code)
}
```

Function `c_init_object_stream` is *synchronous* (this is important!): it spawns `put_object_stream()` or `get_object_stream()` depending on its 2nd argument. Note that `put_object_stream()` still accepts `UnixStream`s in the beta release via parameter `reader` while `get_object_stream()` no longer accepts them via parameter `writer` (called `server` in the code).

So, basically, I am ok with the new design and appreciate it, except that I need the old `get_object_stream()` functionality too.

**Environment**
 - Rust version: [`1.66.1`]
 - lib version [`0.33,0-beta5`]

**Additional context**

The old `get_object_stream()` API is needed to FFI with C/C++.

**Comment by durch** [comment]
@lyokha I think `[get_object_to_writer](https://github.com/durch/rust-s3/blob/363a669232dda87fd819574be16f08542eed3de5/s3/src/bucket.rs#L824)` is what you're after. Underlaying implementation is the same

**Comment by lyokha** [comment]
Thanks! This works. Now I see that the function was just renamed.
